modules/cli/src/main/scala/scala/cli/commands/DependencyUpdate.scala

```diff
@@ -0,0 +1,121 @@
+package scala.cli.commands
+
+import caseapp._
+
+import scala.build.actionable.ActionableDiagnostic.ActionableDependencyUpdateDiagnostic
+import scala.build.actionable.ActionablePreprocessor
+import scala.build.internal.CustomCodeWrapper
+import scala.build.options.Scope
+import scala.build.{CrossSources, Logger, Position, Sources}
+import scala.cli.CurrentParams
+import scala.cli.commands.util.SharedOptionsUtil._
+
+object DependencyUpdate extends ScalaCommand[DependencyUpdateOptions] {
+  override def group                                           = "Main"
+  override def sharedOptions(options: DependencyUpdateOptions) = Some(options.shared)
+
+  def run(options: DependencyUpdateOptions, args: RemainingArgs): Unit = {
+    val verbosity = options.shared.logging.verbosity
+    CurrentParams.verbosity = verbosity
+
+    val inputs       = options.shared.inputsOrExit(args)
+    val logger       = options.shared.logger
+    val buildOptions = options.shared.buildOptions()
+
+    val crossSources =
+      CrossSources.forInputs(
+        inputs,
+        Sources.defaultPreprocessors(
+          buildOptions.scriptOptions.codeWrapper.getOrElse(CustomCodeWrapper),
+          buildOptions.archiveCache,
+          buildOptions.internal.javaClassNameVersionOpt
+        ),
+        logger
+      ).orExit(logger)
+
+    val scopedSources = crossSources.scopedSources(buildOptions).orExit(logger)
+
+    def generateActionableUpdateDiagnostic(scope: Scope)
+      : Seq[ActionableDependencyUpdateDiagnostic] = {
+      val sources = scopedSources.sources(scope, crossSources.sharedOptions(buildOptions))
+
+      if (verbosity >= 3)
+        pprint.err.log(sources)
+
+      val options = buildOptions.orElse(sources.buildOptions)
+      val actionableDiagnostics =
+        ActionablePreprocessor.generateActionableDiagnostics(options).orExit(logger)
```

Comment:
I wonder if we should generate all of the diagnostics here, since there is nothing too specific to `generateActionableDiagnostics` method at this point